config/envoyconfig/listeners.go

```diff
@@ -629,17 +638,15 @@ func (b *Builder) buildRouteConfiguration(name string, virtualHosts []*envoy_con
 func (b *Builder) buildDownstreamTLSContext(ctx context.Context,
 	cfg *config.Config,
 	domain string,
-) *envoy_extensions_transport_sockets_tls_v3.DownstreamTlsContext {
+) (*envoy_extensions_transport_sockets_tls_v3.DownstreamTlsContext, error) {
 	certs, err := cfg.AllCertificates()
 	if err != nil {
-		log.Warn(ctx).Str("domain", domain).Err(err).Msg("failed to get all certificates from config")
-		return nil
+		return nil, fmt.Errorf("get all certificates from config: %w", err)
 	}
 
 	cert, err := cryptutil.GetCertificateForDomain(certs, domain)
 	if err != nil {
-		log.Warn(ctx).Str("domain", domain).Err(err).Msg("failed to get certificate for domain")
-		return nil
+		return nil, fmt.Errorf("failed to get certificate for domain: %w", err)
```

Comment:
This makes the code more restrictive. It may break for existing users.

Reply:
currently it never happens because of 

https://github.com/pomerium/pomerium/blob/9bce8314ba7935df46e7ca4a3a1a958498cc61e3/pkg/cryptutil/tls.go#L58-L61

which I believe is not a correct behaviour, as it anyway results in unusable endpoint. if we would change it, it would just return 404 because cluster would not be created.